### PR TITLE
Minor fixes, mostly posix.c

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/platform/posix.c [hunk 1]
```diff
@@ -229,8 +232,8 @@ bool platform_lock_single_instance()
 	char pidFilePath[MAX_PATH];
 	char separator = platform_get_path_separator();
 
-	strncpy(pidFilePath, _userDataDirectoryPath, MAX_PATH);
-	strncat(pidFilePath, &separator, 1);
+	strncpy(pidFilePath, _userDataDirectoryPath, MAX_PATH - 1);
+	strncat(pidFilePath, &separator, MAX_PATH - strnlen(pidFilePath, MAX_PATH) - 1);
```
Comment (IntelOrca) [inline src/platform/posix.c]: the safe_strncpy and safe_strncat do the ```-1``` for you.

Comment (janisozaur) [inline src/platform/posix.c]: The third parameter is maximum amount of characters to copy, so yes, remaining space. That's a common mistake.

**Comment by janisozaur** [comment]
updated.

**Comment by IntelOrca** [comment]
@janisozaur don't you think its better to use our own safe_strncpy and safe_strncat functions?

**Comment by janisozaur** [comment]
I'm not entirely sold. we have `safe_strncpy`, but not the other one, they tend to hide possible errors. Until we can use `std::string` I'd rather stay with explicit `strncat`.

**Comment by IntelOrca** [comment]
Use safe_strncat, we have that.

**Comment by janisozaur** [comment]
I don't think so: https://github.com/OpenRCT2/OpenRCT2/search?utf8=%E2%9C%93&q=safe_strncat

Do you want to implement one?

**Comment by janisozaur** [comment]
oh, you mean [`safe_strcat`/`safe_strcat_path`](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/util/util.c#L225) in that case they're named improperly, may I rename them?

**Comment by IntelOrca** [comment]
Sorry I called it safe_strcat, I probably thought the n was unnecessary since it had safe before it.

**Comment by IntelOrca** [comment]
I'd rather safe_strncpy was renamed to safe_strcpy.

**Comment by janisozaur** [comment]
While I don't quite agree with you on that @IntelOrca I renamed it how you wanted for sake of consistency.

**Comment by IntelOrca** [comment]
Well its only temporary until we can use the C++ versions I made.

**Pull request merged**